packages/react-data-grid/src/Row.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import rowComparer from './common/utils/RowComparer';
+import Cell from './Cell';
+import { isFrozen } from './ColumnUtils';
+import { Column, RowRenderer, RowRendererProps, CellRenderer, CellRendererProps } from './common/types';
+
+export default class Row extends React.Component<RowRendererProps> implements RowRenderer {
+  static displayName = 'Row';
+
+  static defaultProps = {
+    cellRenderer: Cell,
+    isSelected: false,
+    height: 35
+  };
+
+  private readonly row = React.createRef<HTMLDivElement>();
+  private readonly cells: { [key: string]: CellRenderer | null } = {};
```

Comment:
Maps are a more appropriate data structure for this kind of usage than dynamic objects.
```ts
private readonly cells = new Map<string, CellRenderer>();
```
This should perform much better, and won't lead to js deoptimization.

Reply:
Done